crates/frontend/src/pages/config_version_list.rs

```diff
@@ -0,0 +1,206 @@
+use leptos::*;
+
+use chrono::{NaiveDateTime, TimeZone, Utc};
+use leptos_router::A;
+use serde_json::{json, Map, Value};
+
+use crate::components::skeleton::Skeleton;
+use crate::components::stat::Stat;
+use crate::components::table::types::TablePaginationProps;
+use crate::components::table::{types::Column, Table};
+use crate::types::ListFilters;
+
+use crate::api::fetch_snapshots;
+
+#[component]
+pub fn config_version_list() -> impl IntoView {
+    let tenant_rs = use_context::<ReadSignal<String>>().unwrap();
+
+    // Signals for filters
+    let (filters, set_filters) = create_signal(ListFilters {
+        status: None,
+        from_date: Utc.timestamp_opt(0, 0).single(),
+        to_date: Utc.timestamp_opt(4130561031, 0).single(),
+        page: Some(1),
+        count: Some(10), // Limit of 10 items per page
+    });
+
+    let table_columns = create_memo(move |_| snapshot_table_columns(tenant_rs.get()));
+
+    let snapshots_resource: Resource<(String, i64, i64), Vec<Map<String, Value>>> =
+        create_blocking_resource(
+            move || {
+                (
+                    tenant_rs.get(),
+                    filters.get().page.unwrap_or(1),
+                    filters.get().count.unwrap_or(10),
+                )
+            },
+            |(current_tenant, page, count)| async move {
+                match fetch_snapshots(current_tenant.to_string(), page, count).await {
+                    Ok(response) => response
+                        .data
+                        .into_iter()
+                        .map(|config_version| {
+                            let mut map = Map::new();
+                            map.insert(
+                                "id".to_string(),
+                                Value::String(config_version.id.to_string()),
+                            );
+                            map.insert(
+                                "created_at".to_string(),
+                                json!(config_version.created_at),
+                            );
+                            map.insert("tags".to_string(), json!(config_version.tags));
+                            map
+                        })
+                        .collect(),
```

Comment:
We are loosing on the total_pages and total_items coming in from api. I am ok with the data transformation being done at source, but can you make sure to retain the `total_pages` and `total_items` from the response.